src/routing/bootstrap.rs

```diff
@@ -449,6 +456,13 @@ impl<'a> State<'a> {
                         continue;
                     }
 
+                    if let Some(expected_genesis_key) = expected_genesis_key {
+                        if expected_genesis_key != genesis_key {
+                            error!("Unexpected Genesis key");
```

Comment:
does this have any `real` usage?

Reply:
it's just to make sure we are joining the right network. Currently this check only applies to relocated nodes but it's not hard to extend it to newly joining nodes too. We could for example bundle the expected genesis key in the release binaries, or something like that.